<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>GeometryFactory</author>
 <class>Optimal_transportation</class>
 <widget class="QMainWindow" name="Optimal_transportation">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>646</width>
    <height>703</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Optimal transportation</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources/CGAL.qrc">
    <normaloff>:/cgal/logos/cgal_icon</normaloff>:/cgal/logos/cgal_icon</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout">
    <item row="0" column="0">
     <widget class="QSlider" name="timeSlider">
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="fileToolBar">
   <property name="windowTitle">
    <string>File Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionClear"/>
   <addaction name="actionLoadPoints"/>
   <addaction name="actionSavePoints"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>Visualization Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionInsertPoint"/>
   <addaction name="actionShowRegular"/>
   <addaction name="actionShowPowerdiagram"/>
   <addaction name="actionRecenter"/>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>646</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionClear"/>
    <addaction name="actionLoadPoints"/>
    <addaction name="actionSavePoints"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionInsertRandomPoints"/>
    <addaction name="actionLloydStep"/>
    <addaction name="actionFPAurenhammerStep"/>
    <addaction name="actionShowColoredPowerDiagram"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="actionDisplayGradients"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
  </widget>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="actionAboutCGAL">
   <property name="text">
    <string>About &amp;CGAL</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionInsertRandomPoints">
   <property name="text">
    <string>&amp;Insert random points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actionInsertPoint">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/Input.qrc">
     <normaloff>:/cgal/Input/inputPoint.png</normaloff>:/cgal/Input/inputPoint.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Insert weighted point</string>
   </property>
   <property name="toolTip">
    <string>Insert weighted point</string>
   </property>
   <property name="statusTip">
    <string>Left: Insert vtx</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="resources/File.qrc">
     <normaloff>:/cgal/fileToolbar/fileNew.png</normaloff>:/cgal/fileToolbar/fileNew.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Clear</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionShowPowerdiagram">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/cgal/Optimal_transport_2/Voronoi_diagram_2.png</normaloff>:/cgal/Optimal_transport_2/Voronoi_diagram_2.png</iconset>
   </property>
   <property name="text">
    <string>Show &amp;Voronoi Diagram</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="actionShowRegular">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/cgal/Optimal_transport_2/Delaunay_triangulation_2.png</normaloff>:/cgal/Optimal_transport_2/Delaunay_triangulation_2.png</iconset>
   </property>
   <property name="text">
    <string>Show &amp;Regular Triangulation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionLoadPoints">
   <property name="icon">
    <iconset resource="resources/File.qrc">
     <normaloff>:/cgal/fileToolbar/fileOpen.png</normaloff>:/cgal/fileToolbar/fileOpen.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Load Points...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionSavePoints">
   <property name="icon">
    <iconset resource="resources/File.qrc">
     <normaloff>:/cgal/fileToolbar/fileSave.png</normaloff>:/cgal/fileToolbar/fileSave.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save Points...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionRecenter">
   <property name="icon">
    <iconset resource="resources/Input.qrc">
     <normaloff>:/cgal/Input/zoom-best-fit</normaloff>:/cgal/Input/zoom-best-fit</iconset>
   </property>
   <property name="text">
    <string>Re&amp;center the viewport</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionLloydStep">
   <property name="text">
    <string>Lloyd step</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+L</string>
   </property>
  </action>
  <action name="actionFPAurenhammerStep">
   <property name="text">
    <string>Fixed-point Aurenhammer step</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+A</string>
   </property>
  </action>
  <action name="actionShowColoredPowerDiagram">
   <property name="text">
    <string>Show colored Power diagram</string>
   </property>
  </action>
  <action name="actionDisplayGradients">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Display gradients...</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="resources/Optimal_transportation.qrc"/>
  <include location="resources/File.qrc"/>
  <include location="resources/CGAL.qrc"/>
  <include location="resources/Input.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>timeSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Optimal_transportation</receiver>
   <slot>on_time_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>426</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>422</x>
     <y>-17</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_time_changed(int)</slot>
 </slots>
</ui>
